fedimint-core/src/api.rs

```diff
@@ -905,6 +905,32 @@ pub struct ConsensusContribution {
     pub time: SystemTime,
 }
 
+/// The state of the server returned via APIs
+#[derive(Debug, Clone, Default, Serialize, Deserialize, Eq, PartialEq)]
+pub enum ServerStatus {
+    /// Server needs a password to read configs
+    #[default]
+    AwaitingPassword,
+    /// Waiting for peers to share the config gen params
+    SharingConfigGenParams,
+    /// Ready to run config gen once all peers are ready
+    ReadyForConfigGen,
+    /// We failed running config gen
+    ConfigGenFailed,
+    /// Config is generated, peers should verify the config
+    VerifyingConfigs,
+    /// Restarted from a planned upgrade (requires action to start)
+    Upgrading,
+    /// Consensus is running
+    ConsensusRunning,
+}
+
+#[derive(Debug, Serialize, PartialEq, Eq)]
+pub struct StatusResponse {
+    pub status: ServerStatus,
```

Comment:
The whole struct/endpoint is called "status", so it doesn't seem necessary  to repeat that word?

`server: ServerStatus` and `consensus:Option<ConsensusStatus>`?